x-pack/plugins/security_solution/public/app/routes.tsx

```diff
@@ -22,6 +25,24 @@ interface RouterProps {
 
 const PageRouterComponent: FC<RouterProps> = ({ history, children }) => {
   const dispatch = useDispatch<(action: AppAction) => void>();
+  const {
```

Comment:
I think this whole block should be a hook that we expose out?

Reply:
You should not use `services.ml` here. To access a URL generator try something like 
```typescript
      const urlGenerator = pluginsStart.share.urlGenerators.getUrlGenerator(ML_APP_URL_GENERATOR);
```

Reply:
Btw you can add `ml` as an optional dependency for the plugin.